source/fundamentals/crud/read-operations/distinct.txt

```diff
@@ -0,0 +1,161 @@
+.. _node-fundamentals-distinct:
+
+========================
+Retrieve Distinct Values
+========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Use the ``distinct()`` method to retrieve all distinct values for a specified field 
+across a collection.
+
+Sample Documents
+~~~~~~~~~~~~~~~~
+
+Follow the instructions in the examples below to insert data into a collection and
+return the unique values of a given field. Consider the following collection of 
+documents that describe restaurants: 
+
+.. code-block:: json
+
+   [
+      { "_id": 1, "restaurant": "White Bear", "borough": "Queens", "cuisine": "Chinese" },
+      { "_id": 2, "restaurant": "Via Carota", "borough": "Manhattan", "cuisine": "Italian" },
+      { "_id": 3, "restaurant": "Borgatti's", "borough": "Bronx", "cuisine": "Italian" },
+      { "_id": 4, "restaurant": "Tanoreen", "borough": "Brooklyn", "cuisine": "Middle Eastern" },
+      { "_id": 5, "restaurant": "Äpfel", "borough": "Queens", "cuisine": "German" },
+      { "_id": 6, "restaurant": "Samba Kitchen", "borough": "Manhattan", "cuisine": "Brazilian" },
+   ]
+
+.. include:: /includes/access-cursor-note.rst
+
+Distinct
+--------
+
+Use a document field name as a parameter for the ``distinct()`` method to return a 
+list of the field's unique values. Optionally, you can specify a ``query`` 
+parameter to narrow down your results or an ``options`` parameter to set the 
+string comparison language.
+
+Example
+```````
+
+Some boroughs, like "``Queens``", are represented multiple times in the restaurant 
+documents. However, the following example retrieves only the unique values of the 
+``borough`` field: 
+
+.. code-block:: javascript
+
+   // specify "borough" as the field to return values for
+   const cursor = collection.distinct("borough");
+   await cursor.forEach(console.dir);
+
+This code outputs the following ``borough`` values:
+
+.. code-block:: json
+   :copyable: false
+
+   [ "Bronx", "Brooklyn", "Manhattan", "Queens" ]
+
+Query Parameter
+~~~~~~~~~~~~~~~
+
+You can define a query parameter to prevent the ``distinct()`` method from returning 
+certain distinct values.
+
+Visit :doc:`Specify a Query </fundamentals/crud/query-document>` for more information 
+on constructing a query filter.
+
+Example
+```````
+
+The following example outputs the distinct values of the ``cuisine`` field but 
+excludes restaurants in "``Brooklyn``":
+
+.. code-block:: javascript
+
+   // exclude Brooklyn restaurants from the output
+   const query = { borough: { $ne: "Brooklyn" }};
+
+   // find the filtered distinct values of "cuisine"
+   const cursor = collection.distinct("cuisine", query);
+   await cursor.forEach(console.dir);
+
+In this case, the query filter prevents the ``distinct()`` method from matching a 
+"``Middle Eastern``" Brooklyn restaurant. The code outputs the following ``cuisine`` 
+values:
```

Comment:
issue: The query ignores all documents where `borough` is "Brooklyn", so the byproduct is the omission of "Middle Eastern" from the distinct list of cuisine values since it isn't in any other documents with a different `borough`.